### PR TITLE
mono 6.12.0.206

### PR DESCRIPTION
**Pull request by mchoo7**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/mono.rb [hunk 1]
```diff
@@ -1,8 +1,7 @@
 class Mono < Formula
   desc "Cross platform, open source .NET development framework"
   homepage "https://www.mono-project.com/"
-  url "https://download.mono-project.com/sources/mono/mono-6.12.0.199.tar.xz"
-  sha256 "c0850d545353a6ba2238d45f0914490c6a14a0017f151d3905b558f033478ef5"
+  url "https://github.com/mono/mono.git", using: :git, tag: "mono-6.12.0.206", revision: "0cbf0e290c31adb476f9de0fa44b1d8829affa40"
```
Comment (chenrui333) [inline Formula/m/mono.rb:4]: does upstream stop publishing artifact into `https://download.mono-project.com/sources/mono`?

Comment (fel1x-developer) [reply Formula/m/mono.rb:4]: Not every patch version is released as tarball. 

| Version | Date |
| ------- | ------|
| mono-6.12.0.90 | 2020-09-03 |
| mono-6.12.0.107 | 2020-12-09 |
| mono-6.12.0.122 | 2021-02-22 |
| mono-6.12.0.182 | 2022-06-14 |
| mono-6.12.0.199 | 2023-07-27 |

Thus we need to download from git tag like [arch linux package does](https://gitlab.archlinux.org/archlinux/packaging/packages/mono/-/blob/main/PKGBUILD?ref_type=heads).

Comment (carlocab) [inline Formula/m/mono.rb:4]:
```suggestion
  url "https://github.com/mono/mono.git",
      tag: "mono-6.12.0.206",
      revision: "0cbf0e290c31adb476f9de0fa44b1d8829affa40"
```

File: Formula/m/mono.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 class Mono < Formula
   desc "Cross platform, open source .NET development framework"
   homepage "https://www.mono-project.com/"
-  url "https://download.mono-project.com/sources/mono/mono-6.12.0.199.tar.xz"
-  sha256 "c0850d545353a6ba2238d45f0914490c6a14a0017f151d3905b558f033478ef5"
+  url "https://github.com/mono/mono.git",
+      tag:      "mono-6.12.0.206",
+      revision: "0cbf0e290c31adb476f9de0fa44b1d8829affa40"
```
Comment (ZhongRuoyu) [inline Formula/m/mono.rb:6]: Can we fetch the Git archive instead? A Git clone is usually much heavier and less cache-friendly.

```suggestion
  url "https://github.com/mono/mono/archive/refs/tags/mono-6.12.0.206.tar.gz"
  sha256 "52ca15df4e29d08fbeb7a85bb35db1b2313521999473c5960e50d9bd56389c2e"
```

Comment (fel1x-developer) [reply Formula/m/mono.rb:6]: I already tried that but I couldn't clone the submodules in this way.

**Review by carlocab — APPROVED** [review]
Thanks!

**Review by ZhongRuoyu — APPROVED** [review]

**Pull request merged**